NEA  
https://exoplanetarchive.ipac.caltech.edu/overview/Kepler-20%20b#planet_Kepler-20-b_collapsible  
  
Gautier et al. 2011  
https://ui.adsabs.harvard.edu/abs/2012ApJ...749...15G/abstract  
  
Kepler-20 b  

## Part 1

![title](img/figure3_marked.jpg)

Figure 3 from Griffith et al 2020
https://iopscience.iop.org/article/10.3847/1538-4357/abd6be

In [1]:
## _FeH_ == [Fe/H]
## FeH == Fe/H mol ratio

_FeH_ = 0.00

_MgFe_ = 0.10
_SiFe_ = 0.00
_CaFe_ = 0.00
_AlFe_ = 0.05

_MgH_ = (1+_FeH_) * (1+_MgFe_) - 1
_SiH_ = (1+_FeH_) * (1+_SiFe_) - 1
_CaH_ = (1+_FeH_) * (1+_CaFe_) - 1
_AlH_ = (1+_FeH_) * (1+_AlFe_) - 1

# Sol Abundances, from literature
# https://par.nsf.gov/servlets/purl/10036398
H = 12
Fe = 7.48
Mg = 7.54
Si = 7.52
Ca = 6.33
Al = 6.47

# _MgH_ = log10( MgH / 10^( Mg - H ) )
# MgH = 10^( _MgH_ ) * 10^( Mg - H ))
MgH = 10**( _MgH_ ) * 10**( Mg - H )
print( "[",_MgH_,"]" )

print( MgH ) # Kepler-20 b
print( 10**( Mg - H ) ) # Solar

## Solar Molar Ratio of Fe/H
FeH = 10**( Fe - H )
## Kepler-20 b Molar Ratio
FeH = 10**( _FeH_ ) * 10**( Fe - H )
print(FeH)

# With the Solar Iron abundance we can scale it to the magnesium abundance

## Scaled Fe/Mg Abundance
FeMg = FeH / MgH
print("FeMg:", FeMg)

## Scaled Si/Mg Abundance
SiH = 10**( _SiH_ ) * 10**( Si - H )
SiMg = SiH / MgH
print("SiMg:", SiMg)

## Scaled Si/Mg Abundance
AlH = 10**( _AlH_ ) * 10**( Al - H )
AlMg = AlH / MgH
print("AlMg:", AlMg)

## Scaled Ca/Mg Abundance
CaH = 10**( _CaH_ ) * 10**( Ca - H )
CaMg = CaH / MgH
print("CaMg:", CaMg)

[ 0.10000000000000009 ]
4.3651583224016614e-05
3.467368504525317e-05
3.0199517204020192e-05
FeMg: 0.6918309709189369
SiMg: 0.7585775750291828
AlMg: 0.07585775750291832
CaMg: 0.04897788193684461


## Part 2

Star's age: <8.80 +4.70 -2.70 Gyr

In [2]:
def irradiation_square_m( L_star, orbital_sep, R_planet ):
    import numpy as _np
    
    R_e = 6378 # km
    R_p = R_planet # R_Earth
    R_p *= R_e
    
    L_sun = 3.828E+26 # W
    L_kep = L_star # (log10(solar))
    L_kep = 10**( L_kep ) * L_sun
    
    au_to_km = 1.496e+8 # km
    orbital_sep = orbital_sep # AU
    orbital_sep *= au_to_km
    
    irrad = L_kep/(4 * _np.pi * orbital_sep**2) * (1000)**-2

    return irrad

## Estimates of Irradiance
print( "Kepler-20 b:", round(irradiation_square_m(-0.15, 0.045, 1.91)), "W/m2" )

print( "Mercury:", round(irradiation_square_m(0, 0.4, 0.383)), "W/m2" )

# 1360.8 ± 0.5 W/m2; https://earth.gsfc.nasa.gov/climate/research/solar-radiation
print( "Earth:", round(irradiation_square_m(0, 1, 1)), "W/m2" )

Kepler-20 b: 475854 W/m2
Mercury: 8507 W/m2
Earth: 1361 W/m2


In [3]:
def irradiation( L_star, orbital_sep, R_planet ):
    import numpy as _np
    
    R_e = 6378 # km
    R_p = R_planet # R_Earth
    R_p *= R_e
    
    L_sun = 3.828E+26 # W
    L_kep = L_star # (log10(solar))
    L_kep = 10**( L_kep ) * L_sun
    
    au_to_km = 1.496e+8 # km
    orbital_sep = orbital_sep # AU
    orbital_sep *= au_to_km
    
    irrad = L_kep/(4 * _np.pi * orbital_sep**2) * (_np.pi * R_p**2) # W

    return irrad

## Estimates of Irradiance
print( "Kepler-20 b:", irradiation(-0.15, 0.045, 1.91), "W" )

print( "Mercury:", irradiation(0, 0.4, 0.383), "W" )

print( "Earth:", irradiation(0, 1, 1), "W" )

print( "Irradiance Relative to Earth:", irradiation(-0.15, 0.045, 1.91)/irradiation(0, 1, 1) )

Kepler-20 b: 2.218500278023708e+20 W
Mercury: 1.5947600430548e+17 W
Earth: 1.7394733544353568e+17 W
Irradiance Relative to Earth: 1275.3861807465548


![title](img/fulton2017_.png)

## Part 3: 

M_Earth values from Gautier et al. 2011: 8.7  +2.1  -2.2  
R_Earth values from Gautier et al. 2011: 1.91 +0.12 -0.21

Using scaled stellar abundances but leaving default core/mantle composition

python example_by_mass_with_flags.py --mass=8.7 --FeMg=0.69 --SiMg=0.76 --AlMg=0.08 --CaMg=0.05  
  
Mass =  8.700 Earth masses  
Radius =  1.79655 Earth radii  
Density =  8.269 g/cc  
Core Mass Fraction =  29.33  
Core Radius Fraction =  48.02  
CMB Pressure =  1265.00 GPa  
CMB Temperature =  5087.98 K  
number of oceans: 0.00  
Central core pressure 3.52 TPa  
Central core Temperature 7071.38 K  

python example_by_mass_with_flags.py --mass=10.8 --FeMg=0.69 --SiMg=0.76 --AlMg=0.08 --CaMg=0.05  
  
Mass =  10.800 Earth masses  
Radius =  1.89248 Earth radii  
Density =  8.782 g/cc  
Core Mass Fraction =  29.33  
Core Radius Fraction =  47.91  
CMB Pressure =  1609.00 GPa  
CMB Temperature =  5285.28 K  
number of oceans: 0.00  
Central core pressure 4.46 TPa  
Central core Temperature 7271.56 K  

python example_by_mass_with_flags.py --mass=6.5 --FeMg=0.69 --SiMg=0.76 --AlMg=0.08 --CaMg=0.05  
  
Mass =  6.500 Earth masses  
Radius =  1.67092 Earth radii  
Density =  7.679 g/cc  
Core Mass Fraction =  29.33  
Core Radius Fraction =  48.23  
CMB Pressure =  924.00 GPa  
CMB Temperature =  4644.50 K  
number of oceans: 0.00  
Central core pressure 2.58 TPa  
Central core Temperature 6511.50 K  

## Part 4: Adjusting Core and Mantle Composition

M_Earth values from Gautier et al. 2011: 8.7  +2.1  -2.2  
R_Earth values from Gautier et al. 2011: 1.91 +0.12 -0.21

![title](img/berf.jpg)

^^ plot above ^^
  
python example_by_mass_with_flags.py --mass=8.7 --FeMg=0.69 --SiMg=0.76 --AlMg=0.08 --CaMg=0.05 --mol_frac_Fe_mantle=0.0 --wt_frac_Si_core=0.3 --wt_frac_O_core=0.1  
  
Mass =  8.700 Earth masses  
Radius =  1.91089 Earth radii  
Density =  6.872 g/cc  
Core Mass Fraction =  40.89  
Core Radius Fraction =  66.55  
CMB Pressure =  560.00 GPa  
CMB Temperature =  3940.96 K  
number of oceans: 0.00  
Central core pressure 1.48 TPa  
Central core Temperature 7281.13 K  

Upper Mass; 0.10 radius discrepency with 8.7 mass; within radius error  
  
python example_by_mass_with_flags.py --mass=10.8 --FeMg=0.69 --SiMg=0.76 --AlMg=0.08 --CaMg=0.05 --mol_frac_Fe_mantle=0.0 --wt_frac_Si_core=0.3 --wt_frac_O_core=0.1  
  
Mass =  10.800 Earth masses  
Radius =  2.01787 Earth radii  
Density =  7.245 g/cc  
Core Mass Fraction =  40.89  
Core Radius Fraction =  66.37  
CMB Pressure =  701.00 GPa  
CMB Temperature =  4242.99 K  
number of oceans: 0.00  
Central core pressure 1.85 TPa  
Central core Temperature 7783.63 K  

Lower Mass; 0.13 radius discrepency with 8.7 mass; within radius error  
  
python example_by_mass_with_flags.py --mass=6.5 --FeMg=0.69 --SiMg=0.76 --AlMg=0.08 --CaMg=0.05 --mol_frac_Fe_mantle=0.0 --wt_frac_Si_core=0.3 --wt_frac_O_core=0.1  
  
Mass =  6.500 Earth masses  
Radius =  1.77317 Earth radii  
Density =  6.426 g/cc  
Core Mass Fraction =  40.89  
Core Radius Fraction =  66.82  
CMB Pressure =  416.00 GPa  
CMB Temperature =  3578.16 K  
number of oceans: 0.00  
Central core pressure 1.09 TPa  
Central core Temperature 6644.97 K  

Earth for comparison
![title](img/earth.jpg)

## Part 5: Mineralogy